src/VisualStudio/Core/Def/LanguageService/AbstractPackage`2.cs

```diff
@@ -64,6 +51,17 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
             return _languageService.ComAggregate;
         });
 
+        await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
+        var shell = (IVsShell7)await GetServiceAsync(typeof(SVsShell)).ConfigureAwait(true);
+        cancellationToken.ThrowIfCancellationRequested();
+        Assumes.Present(shell);
```

Comment:
could this move below the foreach?  is that beneficial?  my thinking is smiply that we don't acutaly need this service until later, so why not delay till then?

Reply:
It can, there is no benefit to it being here. This is effectively a synchronous call anyway because SVsShell is always existing sync service, it never yields.

Reply:
Because Roslyn has their [own version](https://github.com/dotnet/roslyn/issues/73636) of these extensions, I would love to be able to replace all three lines with just:

`var shell = this.GetServiceAsync<SVsShell, IVsShell>(cancellationToken)`